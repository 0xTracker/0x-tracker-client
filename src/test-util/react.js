import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
