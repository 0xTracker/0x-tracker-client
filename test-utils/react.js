import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
