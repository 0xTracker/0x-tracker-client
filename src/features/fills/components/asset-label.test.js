import React from 'react';

import { renderWithRouter } from '../../../test-util/react';
import AssetLabel from './asset-label';

describe('asset label component', () => {
  it('should render for ERC-20 asset', () => {
    const { container } = renderWithRouter(
      <AssetLabel
        asset={{
          tokenAddress: '0x12345',
          tokenSymbol: 'WETH',
          tokenType: 'Wrapped Ether',
          type: 'erc-20',
        }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render for ERC-721 asset', () => {
    const { container } = renderWithRouter(
      <AssetLabel
        asset={{
          tokenAddress: '0x12345',
          tokenId: 999,
          tokenSymbol: 'CK',
          tokenType: 'CryptoKitties',
          type: 'erc-721',
        }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render for unknown token', () => {
    const { container } = renderWithRouter(
      <AssetLabel
        asset={{
          tokenAddress: '0x12345',
          tokenId: 999,
          type: 'erc-721',
        }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
