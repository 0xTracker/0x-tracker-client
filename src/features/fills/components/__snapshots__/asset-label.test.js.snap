// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asset label component should render for ERC-20 asset 1`] = `
<a
  class="link__InternalLink-sc-1m1n17q-1 IiDhz"
  href="/tokens/0x12345"
>
  Wrapped Ether
   
   
  (
  WETH
  )
</a>
`;

exports[`asset label component should render for ERC-721 asset 1`] = `
<a
  class="link__OutboundLink-sc-1m1n17q-0 iAzcTd"
  href="https://opensea.io/assets/0x12345/999"
  rel="noopener noreferrer"
  target="_blank"
>
  CryptoKitties
   
  #
  999
   
  (
  CK
  )
</a>
`;

exports[`asset label component should render for unknown token 1`] = `
<a
  class="link__OutboundLink-sc-1m1n17q-0 iAzcTd"
  href="https://opensea.io/assets/0x12345/999"
  rel="noopener noreferrer"
  target="_blank"
>
  Unknown
   
  #
  999
   
</a>
`;
